website/src/content/main-concepts/user-permissions.mdx

```diff
@@ -2,10 +2,219 @@
+
+```jsx
+import { PERMISSIONS } from './constants/permissions.js';
+
+const ExampleComponent = () => {
+  const canManageProducts = useIsAuthorized({
+    demandedPermissions: [PERMISSIONS.ManageProducts],
+  });
+
+  return <span>{ canManageProducts ? 'Authorized' : 'Not authorized' }</span>;
+};
+```
+
+## `injectAuthorized(permissions, [options])`
+
+A React higher-order component that evaluates the demanded permissions against the user permissions and injects the result as a component prop (default to `isAuthorized`).
+
+### Supported arguments
+
+- `demandedPermissions`: an array of demanded [general permissions](#general-permissions).
+- `options` (_optional_)
+  - `actionRights`: (_optional_) an array of [action rights](#action-rights).
+- `propName`: (_optional_) the name of the injected prop (default to `isAuthorized`).
+
+### Usage
+
+```jsx
+const InputField = props => (
+  <Input
+    disabled={!props.isAuthorized}
+  />
+);
+
+export default injectAuthorized([PERMISSIONS.ViewProducts])(
+  InputField
+);
+```
+
+## `<RestrictedByPermissions>`
+
+A React component that evaluates the demanded permissions against the user permissions and renders the result as a function with the a named argument `isAuthorized`.
+match, otherwise the passed fallback component.
+
+### Supported props
+
+- `permissions`: an array of demanded [general permissions](#general-permissions).
+- `actionRights`: (_optional_) an array of demanded [action rights](#action-rights).
+- `unauthorizedComponent`: (_optional_) a React component instance to be rendered in case the evaluation of the permissions returns `false`.
+  rendered in case the permissions do not match
+- `render`: (_optional_) a render prop function that should return a React node. The function is called with the named boolean argument `isAuthorized`.
+  rendered in case the permissions match
+  - Note: when a function is passed it gets invoked with an object
+    containing `isAuthorized` for example `({ isAuthorized }) => <div />`
+- `children`: (_optional_) a function returning an React element or a node to be
+  rendered in case the permissions match
+  - Note: when a function is passed it is invoked with an object
+    containing `isAuthorized` for example `({ isAuthorized }) => <div />`
+
+### Usage
+
+```jsx
+<RestrictedByPermissions permissions={[PERMISSIONS.ManageOrders]} unauthorizedComponent={Unauthorized}>
+  <MyAuthorizedComponent />
+</RestrictedByPermissions>
+
+<RestrictedByPermissions permissions={[PERMISSIONS.ManageOrders]}
+  {({ isAuthorized }) => <button disabled={!isAuthorized}>My button</button>}
+</RestrictedByPermissions>
+```
+
+# Best practices
+
+## Using a file with constants
+
+It is recommended to store general permissions and action rights used by your application into a `permissions.js` file. This helps to avoid repeating permission names across your code base. Preventing typographical errors while giving an overview of used permissions in an application easily.
```

Comment:
```suggestion
It is recommended to store general permissions and action rights used by your application into a `permissions.js` file. This helps to avoid repeating permission names across your code base and preventing typographical errors while giving an overview of used permissions in an application easily.
```